src/main/java/org/folio/service/orders/OrderValidationService.java

```diff
@@ -0,0 +1,364 @@
+   * Checks PO Number presence, validates that provided order id corresponds to one set in order and its lines.
+   * @param orderId Purchase Order id
+   * @param compPO Purchase Order to validate
+   * @return completable future which might be completed with {@code true} if order is valid, {@code false} if not valid or an exception if processing fails
+   */
+  public Future<List<Error>> validateOrderForPut(String orderId, CompositePurchaseOrder compPO,
+      RequestContext requestContext) {
+    // The PO Number is required for existing orders
+    List<Error> resultErrors = new ArrayList<>();
+    if (StringUtils.isEmpty(compPO.getPoNumber())) {
+      resultErrors.add(ErrorCodes.PO_NUMBER_REQUIRED.toError());
+    }
+
+    // Validate order uuid
+    if (!compPO.getId().equals(orderId)) {
+      resultErrors.add(ErrorCodes.MISMATCH_BETWEEN_ID_IN_PATH_AND_BODY.toError());
+    } else if (isNotEmpty(compPO.getCompositePoLines())) {
+      // Validate that each PO Line has correct order id
+      compPO.getCompositePoLines().forEach(poLine -> {
+        if (!orderId.equals(poLine.getPurchaseOrderId())) {
+          Error error = ErrorCodes.MISMATCH_BETWEEN_ID_IN_PATH_AND_BODY.toError();
+          if (StringUtils.isNotEmpty(poLine.getPoLineNumber())) {
+            error.getParameters()
+              .add(new Parameter().withKey(PO_LINE_NUMBER)
+                .withValue(poLine.getPoLineNumber()));
+          }
+          resultErrors.add(error);
+        }
+      });
+    }
+
+    return configurationEntriesCache.loadConfiguration(ORDER_CONFIG_MODULE_NAME, requestContext)
+      .compose(tenantConfig -> validateOrderForPost(compPO, tenantConfig, requestContext))
+      .map(errors -> {
+        resultErrors.addAll(errors);
+        return resultErrors;
+      });
+  }
+
+  public Future<Void> validateOrderForUpdate(CompositePurchaseOrder compPO, CompositePurchaseOrder poFromStorage,
+      boolean deleteHoldings, RequestContext requestContext) {
+    logger.info("validateOrderForUpdate :: orderId: {}", compPO.getId());
+    return validateAcqUnitsOnUpdate(compPO, poFromStorage, requestContext)
+      .compose(ok -> prefixService.validatePrefixAvailability(compPO.getPoNumberPrefix(), requestContext))
+      .compose(ok -> suffixService.validateSuffixAvailability(compPO.getPoNumberSuffix(), requestContext))
+      .compose(ok -> poNumberHelper.validatePoNumberPrefixAndSuffix(compPO))
+      .compose(ok -> poNumberHelper.validatePoNumber(poFromStorage, compPO, requestContext))
+      .compose(ok -> {
+        if (isTransitionToApproved(poFromStorage, compPO)) {
+          return checkOrderApprovalPermissions(compPO, requestContext);
+        }
+        return Future.succeededFuture();
+      })
+      .compose(ok -> {
+        if (isTransitionToPending(poFromStorage, compPO)) {
+          checkOrderUnopenPermissions(requestContext);
+          return unOpenCompositeOrderManager.process(compPO, poFromStorage, deleteHoldings, requestContext);
```

Comment:
adding dependency of unOpenCompositeOrderManager and invoking logic to process makes our validation service not lightweight, need to think how to avoid this dependency if possible

Reply:
Yes, more refactoring is needed.

Reply:
if you prefer to refactor in subsequent Pr I will resolve this comment

Reply:
I think this should not be part of MODORDERS-1098 because it is unrelated. Could you create a new ticket ?